shared/js/gzip/gzip.js

```diff
@@ -0,0 +1,39 @@
+/* exported Gzip */
+'use strict';
+
+var Gzip = {
+
+  WORKER_URL: 'shared/js/gzip/gzip_worker.js',
+  compress: function(payload) {
+
+    var self = this;
+    return new Promise(function(resolve, reject) {
+
+      var worker = new Worker(self.WORKER_URL);
```

Comment:
gzip is a singleton, so I think you can just say gzip.WORKER_URL here.  It means you don't need the self thing at all.